AK/MemoryStream.h

```diff
@@ -34,6 +34,29 @@ class FixedMemoryStream : public SeekableStream {
     size_t offset() const;
     size_t remaining() const;
 
+    /// Read a value, but referring to the stream's underlying data instead of copying it.
```

Comment:
FixedMemoryStream can be read-only - should we keep the user from obtaining a non-const reference if it is?

I'd imagine something like the following:

```cpp
if constexpr (!IsConst<T>) {
    if (!m_writing_enabled)
        return Error::from_string_view("Tried to obtain a non-const reference from a read-only FixedMemoryStream");
}
```

Reply:
This requires some ugliness in BitmapFont because its fields have to be mutable spans, but that's a future fixme.